### ISSUE TITLE
The test `Congruence 029` sometimes fails (in the CI at least)

### ISSUE THREAD

**Issue by james-d-mitchell** [libsemigroups/libsemigroups#566]
~~~
test_all is a Catch v2.13.10 host application.
Run with -? for options

-------------------------------------------------------------------------------
Congruence 029: left congruence on finite semigroup
-------------------------------------------------------------------------------
tests/test-cong.cpp:806
...............................................................................

tests/test-cong.cpp:832: FAILED:
  REQUIRE( cong.contains(100101_w, 0010001_w) )
with expansion:
  false

............
~~~

for example in:

https://dev.azure.com/jdm3/jdm3/_build/results?buildId=3618&view=logs&jobId=35ee7e5e-1046-5c42-21f6-afda6bc5c21b&j=35ee7e5e-1046-5c42-21f6-afda6bc5c21b&t=caebb6af-0611-55eb-e715-38ee266099eb

**Comment by james-d-mitchell** [comment]
Another one:

https://dev.azure.com/jdm3/jdm3/_build/results?buildId=3637&view=logs&jobId=35ee7e5e-1046-5c42-21f6-afda6bc5c21b&j=35ee7e5e-1046-5c42-21f6-afda6bc5c21b&t=caebb6af-0611-55eb-e715-38ee266099eb

**Comment by Joseph-Edwards** [comment]
Another:
https://dev.azure.com/jdm3/jdm3/_build/results?buildId=3649&view=logs&jobId=35ee7e5e-1046-5c42-21f6-afda6bc5c21b&j=35ee7e5e-1046-5c42-21f6-afda6bc5c21b&t=caebb6af-0611-55eb-e715-38ee266099eb

@james-d-mitchell hypothesised that this happens because `KnuthBendix` now sometimes wins (and is wrong). I recreated this test, both with `KnuthBendix<RewriteTrie>` and `KnuthBendix<RewriteFromLeft>` and both fail in the same place.

**Comment by Joseph-Edwards** [comment]
The below tests fail, but probably shouldn't:
```cpp
LIBSEMIGROUPS_TEST_CASE("KnuthBendix",
                          "998",
                          "left congruence on finite semigroup",
                          "[quick]") {
    auto                  rg = ReportGuard(false);
    FroidurePin<Transf<>> S;
    S.add_generator(Transf<>({1, 3, 4, 2, 3}));
    S.add_generator(Transf<>({3, 2, 1, 3, 3}));

    auto l = 010001100_w;
    auto r = 10001_w;

    word_type w3 = S.factorisation(Transf<>({1, 3, 1, 3, 3}));
    word_type w4 = S.factorisation(Transf<>({4, 2, 4, 4, 2}));

    KnuthBendix<detail::RewriteTrie> kb(left, to_presentation<word_type>(S));
    kb.add_pair(l, r);
    REQUIRE(kb.contains(100101_w, 0010001_w));
  }

  LIBSEMIGROUPS_TEST_CASE("KnuthBendix",
                          "999",
                          "left congruence on finite semigroup",
                          "[quick]") {
    auto                  rg = ReportGuard(false);
    FroidurePin<Transf<>> S;
    S.add_generator(Transf<>({1, 3, 4, 2, 3}));
    S.add_generator(Transf<>({3, 2, 1, 3, 3}));

    auto l = 010001100_w;
    auto r = 10001_w;

    word_type w3 = S.factorisation(Transf<>({1, 3, 1, 3, 3}));
    word_type w4 = S.factorisation(Transf<>({4, 2, 4, 4, 2}));

    KnuthBendix<detail::RewriteFromLeft> kb(left,
                                            to_presentation<word_type>(S));
    kb.add_pair(l, r);
    REQUIRE(kb.contains(100101_w, 0010001_w));
  }
```
(Tests inspired by `Congruence 029` but with `Congruence` -> `KnuthBendix`)

**Comment by james-d-mitchell** [comment]
Just to update, I know how to fix this, there was a bug in the `equal_to` method of `KnuthBendix` for 1-sided congruences. Preparing a fix.

**Comment by Joseph-Edwards** [comment]
Fixed in #586
